geospatial/src/Dem_TEST.cc

```diff
@@ -245,12 +257,28 @@ TEST_F(DemTest, NonEarthDem)
   // unable to get coordinates in WGS84
   gz::math::Angle latitude, longitude;
   EXPECT_FALSE(dem.GeoReferenceOrigin(latitude, longitude));
+}
 
-  // The Load() method in Dem.cc should set the
-  // isNonEarthDEM flag.
-  EXPECT_TRUE(dem.GetNonEarthDEM());
-
-  // This flag can be overridden externally.
-  dem.SetNonEarthDEM(false);
-  EXPECT_FALSE(dem.GetNonEarthDEM());
+TEST_F(DemTest, LunarDemLoad)
+{
+  // Load Moon DEM
+  common::Dem dem;
+  auto path = common::testing::TestFile("data", "dem_moon.tif");
+  // Providing spherical coordinates object.
+  auto moonSc = math::SphericalCoordinates(
+      math::SphericalCoordinates::MOON_SCS);
+  dem.SetSphericalCoordinates(moonSc);
+  EXPECT_EQ(dem.Load(path), 0);
+  EXPECT_NEAR(dem.WorldWidth(), 80.0417, 1e-2);
+  EXPECT_NEAR(dem.WorldHeight(), 80.0417, 1e-2);
+
+  // Use custom spherical coordinates object with same axes as the moon.
+  auto customSc = math::SphericalCoordinates(
+      math::SphericalCoordinates::CUSTOM_SURFACE,
+      moonSc.SurfaceRadius(),
+      moonSc.SurfaceRadius());
```

Comment:
This would make more sense to me:
```suggestion
      moonSc.SurfaceAxisEquatorial(),
      moonSc.SurfaceAxisPolar());
```
What do you think?

Reply:
Changed : https://github.com/gazebosim/gz-common/pull/377/commits/870144e878127e90613987ded10691ca3e84a98a, also updated the transformation methods to use equatorial and polar axes.